pkg/core/proxy/integrations/http/match.go

```diff
@@ -20,23 +25,33 @@ func match(ctx context.Context, logger *zap.Logger, matchParams *matchParams, mo
 		case <-ctx.Done():
 			return false, nil, ctx.Err()
 		default:
-			tcsMocks, err := mockDb.GetFilteredMocks()
+			unfilteredMocks, err := mockDb.GetUnFilteredMocks()
 
 			if err != nil {
-				utils.LogError(logger, err, "failed to get tcs mocks")
+				utils.LogError(logger, err, "failed to get unfilteredMocks mocks")
 				return false, nil, errors.New("error while matching the request with the mocks")
 			}
+
+			logger.Debug(fmt.Sprintf("Length of unfilteredMocks:%v", len(unfilteredMocks)))
+
 			var eligibleMocks []*models.Mock
 
-			for _, mock := range tcsMocks {
+			for _, mock := range unfilteredMocks {
 				if ctx.Err() != nil {
 					return false, nil, ctx.Err()
 				}
 				if mock.Kind == models.HTTP {
```

Comment:
Unnecessary branching. You should rather check if its not http and then skip that.